core/deployment/src/main/java/io/quarkus/deployment/dev/testing/HtmlAnsiProcessor.java

```diff
@@ -0,0 +1,141 @@
+
+    private final List<String> closingAttributes = new ArrayList<String>();
+    private boolean concealOn = false;
+
+    public HtmlAnsiProcessor(OutputStream os) {
+        super(os);
+    }
+
+    @Override
+    protected void processSetAttribute(int attribute) throws IOException {
+        switch (attribute) {
+            case ATTRIBUTE_CONCEAL_ON:
+                write("\u001B[8m");
+                concealOn = true;
+                break;
+            case ATTRIBUTE_INTENSITY_BOLD:
+                writeAttribute("b");
+                break;
+            case ATTRIBUTE_INTENSITY_NORMAL:
+                closeAttributes();
+                break;
+            case ATTRIBUTE_UNDERLINE:
+                writeAttribute("u");
+                break;
+            case ATTRIBUTE_UNDERLINE_OFF:
+                closeAttributes();
+                break;
+            case ATTRIBUTE_NEGATIVE_ON:
+                break;
+            case ATTRIBUTE_NEGATIVE_OFF:
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    protected void processAttributeReset() throws IOException {
+        if (concealOn) {
+            write("\u001B[0m");
+            concealOn = false;
+        }
+        closeAttributes();
+    }
+
+    @Override
+    protected void processSetForegroundColor(int color, boolean bright) throws IOException {
+        writeAttribute("span style=\"color: " + ANSI_COLOR_MAP[color] + ";\"");
+    }
+
+    @Override
+    protected void processSetBackgroundColor(int color, boolean bright) throws IOException {
+        writeAttribute("span style=\"background-color: " + ANSI_COLOR_MAP[color] + ";\"");
+    }
+
+    @Override
+    protected void processSetForegroundColorExt(int r, int g, int b) throws IOException {
+        writeAttribute(
+                "span style=\"color: #" + Integer.toHexString(r) + Integer.toHexString(g) + Integer.toHexString(b) + ";\"");
+    }
+
+    @Override
+    protected void processSetBackgroundColorExt(int r, int g, int b) throws IOException {
+        writeAttribute("span style=\"background-color: #" + Integer.toHexString(r) + Integer.toHexString(g)
+                + Integer.toHexString(b) + ";\"");
+    }
+
+    @Override
+    protected void processSetForegroundColorExt(int paletteIndex) throws IOException {
+        writeAttribute("span style=\"color: " + COLORS[paletteIndex] + ";\"");
+    }
+
+    @Override
+    protected void processSetBackgroundColorExt(int paletteIndex) throws IOException {
+        writeAttribute("span style=\"background-color: " + COLORS[paletteIndex] + ";\"");
+    }
+
+    private void write(String s) throws IOException {
+        super.os.write(s.getBytes());
+    }
+
+    private void writeAttribute(String s) throws IOException {
+        write("<" + s + ">");
+        closingAttributes.add(0, s.split(" ", 2)[0]);
+    }
+
+    private void closeAttributes() throws IOException {
```

Comment:
Unfortunately, we're not really able to close the attributes properly with the new jansi version... but apparently we weren't closing things properly in the past so it shouldn't change a thing.